sys-process/rtirq/rtirq-20191121.ebuild

```
@@ -0,0 +1,41 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit linux-info systemd
+
+DESCRIPTION="Modify realtime scheduling policy and priority of IRQ handlers"
+HOMEPAGE="https://www.rncbc.org/archive/#rtirq"
+SRC_URI="https://www.rncbc.org/archive/${P}.tar.gz
+	https://www.rncbc.org/archive/old/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+src_prepare() {
+	get_version
+
+	# Correct config file path.
+	sed -i -e "s:^\(RTIRQ_CONFIG\=\)\(.*\):\1/etc/conf.d/rtirq:" ${PN}.sh || die
+	sed -i -e "s:/etc/sysconfig/rtirq:/etc/conf.d/rtirq:" ${PN}.conf || die
+	sed -i -e "s:/etc/init.d/rtirq:/usr/bin/rtirq:" ${PN}{,-resume}.service || die
+
+	default
+}
+
+src_install() {
+	newbin rtirq.sh rtirq
+	newinitd "${FILESDIR}"/rtirq.initd rtirq
+	newconfd rtirq.conf rtirq
+	systemd_dounit rtirq{,-resume}.service
+}
+
+pkg_postinst() {
+	if [[ ${KV_LOCAL} != *rt* ]] && ! grep -qw threadirqs /proc/cmdline ; then
```

`$(grep -qw threadirqs /proc/cmdline || die)` should work, could you try?

---

ebuild shouldn't die if there is no `threadirqs` in `/proc/cmdline`. What I can do is to read `/proc/cmdline` into variable first, something like:
```suggestion
	local cmdline=$(< /proc/cmdline) || die
	if [[ ${KV_LOCAL} != *rt* && ${cmdline} != *threadirqs* ]] ; then
```
But I'm not sure if this complication is needed here. We just deciding whether we need to show postinstall message or not.

---

`grep` itself can fail, even if the file exists. Hence we try to guard it. 

But looking at what this even does, would you consider using `CONFIG_CHECK` instead?

---

Hrmm.. I just realized, that `threadirqs` parameter only works when `CONFIG_IRQ_FORCED_THREADING` is enabled. So `CONFIG_CHECK` is indeed needed here. But we still need to check that `threadirqs` is passed as a kernel cmdline parameter. Also instead of checking `${KV_LOCAL}` it's better to check for `CONFIG_PREEMPT_RT` instead.